configs/arc64.config

```diff
@@ -0,0 +1,54 @@
+CT_CONFIG_VERSION="3"
+CT_EXPERIMENTAL=y
+CT_LOCAL_TARBALLS_DIR="${CT_PREFIX:-${HOME}/x-tools}/sources"
+# CT_PREFIX_DIR_RO is not set
+CT_PATCH_BUNDLED_LOCAL=y
+CT_LOCAL_PATCH_DIR="${CT_TOP_DIR}/../../patches"
+# CT_LOG_PROGRESS_BAR is not set
+CT_ARCH_ARC=y
+CT_ARCH_64=y
+CT_TARGET_VENDOR="zephyr"
+CT_BINUTILS_SRC_DEVEL=y
+CT_BINUTILS_DEVEL_URL="https://github.com/foss-for-synopsys-dwc-arc-processors/binutils-gdb.git"
+CT_BINUTILS_DEVEL_REVISION="9742c8bf3899b833122ce0bfbf3e112132c907cb"
+CT_NEWLIB_SRC_DEVEL=y
+CT_NEWLIB_DEVEL_URL="https://github.com/foss-for-synopsys-dwc-arc-processors/newlib.git"
+CT_NEWLIB_DEVEL_REVISION="850160f6ba8620a6841ce956ee6c88ecc73399ac"
```

Comment:
We need the patches for newlib to be against https://github.com/zephyrproject-rtos/newlib-cygwin

Reply:
Same here. I may generate patches, but see my explanation above for GCC.

Reply:
We need to utilize the same source base for ALL the architectures as much as possible.  We can make an exception for GCC, but see my comment above about having a snapshot of upstream gcc sources + patches on top of that for any additional ARC support.

Reply:
Well I think there're many reasons to use the same versions of all components for all architectures in Zephyr. But I'm not sure how feasible that could be? Some need more recent version due to significant improvements & fixes, some prefer to stick to older versions because they are known to work.

And I think all becomes much worse if we decide to use a snapshot of not yet released components. Maybe we need to discuss that with wider audience?

As for the current ARC64 changes I think it's fine to differ from other "more stable/traditional" arches becase:
1. We may not introduce additional turbulence for other arches switching all to WIP components.
2. We'll minimize maintenance - only update commit hashes from time to time instead of huge patches
3. ARC64 will also act as a guinea pig highlighting issues we'll see when we eventually switch to GCC 11, Binutils 2.37 etc
4. This all is for a limited amount of time - we're interested in upstreaming GCC/Binutils/Newlib ASAP. It's too late for GCC 11 obviously, but GCC 12 might have if not all but at least some ARC64 support I hope.

Reply:
@nashif can chime in and we can discuss in the toolchain working group.

However, I understand what you are saying from a toolchain point of view, but from the Zephyr point of view this can be a huge pain.  I'm less concerned about gcc/binutils and more about newlib.  As I mentioned on a call we had huge pains when the xtensa newlib was out of sync with all the other zephyr arch's.  So we really need newlib to be in sync.

right now I'd lean to saying NOT to include the ARC64 support in the SDK and just have the Zephyr ARC64 support point at an "external" toolchain.  (similar to how esp32 works).

Reply:
I guess Newlib is the simplest thing among these 3 (GCC, Binutils & Newlib) as it's the smallest code-base and amount of changes is not that huge. Moreover for now we use not very recent Newlib (like v3.0 or v3.1) as a base so it's super easy to generate a cumulative patch for ARC64 and rarely update it as needed. As well as re-base it on top of v3.3 (currently used in Zephyr) or even more up-to-date versions when time comes.

So think again, maybe accept ARC64 SDK ;)